### ISSUE TITLE
Pycolmap problem on some function

### ISSUE THREAD

**Issue by yyfz** [colmap/colmap#2937]
**Describe the bug**

Hi, thanks for your great work! I install pycolmap from PyPI's wheel using `pip install pycolmap==3.10.0 pyceres` but I encountered the following issue:

```python
Python 3.10.0 | packaged by conda-forge | (default, Nov 20 2021, 02:24:10) [GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pycolmap
>>> ba_options_tmp = pycolmap.BundleAdjustmentOptions()
>>> ba_options_tmp.solver_options
TypeError: Unregistered type : ceres::Solver::Options

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: Unable to convert function return value to a Python type! The signature was
        (self: pycolmap.BundleAdjustmentOptions) -> ceres::Solver::Options
>>> ba_options_tmp.solver_options.function_tolerance *= 10
TypeError: Unregistered type : ceres::Solver::Options

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: Unable to convert function return value to a Python type! The signature was
        (self: pycolmap.BundleAdjustmentOptions) -> ceres::Solver::Options 
>>> import pyceres
>>> pycolmap.__version__
'3.10.0'
>>> pyceres.__version__
'2.3'
```

**Comment by ahojnnes** [comment]
With pycolmap 3.10, you need to explicitly import pyceres before accessing pycolmap members depending on ceres. The following should work:
```
>>> import pycolmap
>>> import pyceres
>>> ba_options_tmp = pycolmap.BundleAdjustmentOptions()
>>> ba_options_tmp.solver_options
<pyceres.SolverOptions object at 0x107f9a170>
```

This should not be necessary with the current development version and the future 3.11 release.

**Comment by yyfz** [comment]
Sorry, I try and I get the same error:

```python
Python 3.9.20 | packaged by conda-forge | (main, Sep 30 2024, 17:49:10) 
[GCC 13.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pycolmap
>>> import pyceres
>>> ba_options_tmp = pycolmap.BundleAdjustmentOptions()
>>> ba_options_tmp.solver_options
TypeError: Unregistered type : ceres::Solver::Options

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: Unable to convert function return value to a Python type! The signature was
        (self: pycolmap.BundleAdjustmentOptions) -> ceres::Solver::Options
>>> pycolmap.__version__
'3.10'
>>> pyceres.__version__
'2.2'
>>> 
```

**Comment by sarlinpe** [comment]
If you used wheels from PyPI, I think that pycolmap 3.10 requires pyceres 2.3 since the 2.2 wheels are based on the older CentOS-based CI. It's quite annoying to keep track of the pycolmap - pyceres dependencies, we should improve this - for example by automatically flagging when the two packages are built with different environments.

**Comment by yyfz** [comment]
@ahojnnes @sarlinpe I resolved the issue by switching to pyceres version 2.3. Thank you for your quick response!

**Comment by PenghaoYin** [comment]
I came across with the same problem when I was testing vggsfm. I tried many ways in vain, but this solution saved me! Thank all of you in this conversation!

**Comment by jxhoh** [comment]
>>> import pycolmap
>>> import pyceres
>>> ba_options_tmp = pycolmap.BundleAdjustmentOptions()
>>> ba_options_tmp.solver_options
TypeError: Unregistered type : ceres::Solver::Options

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: Unable to convert function return value to a Python type! The signature was
        (self: pycolmap.BundleAdjustmentOptions) -> ceres::Solver::Options
>>> try:
...     # 优先使用 importlib.metadata（Python 3.8+）
...     from importlib.metadata import version
...     pyceres_version = version('pyceres')
... except ImportError:
...     # 如果 Python 版本低于 3.8，使用 pkg_resources
...     from pkg_resources import get_distribution
...     pyceres_version = get_distribution('pyceres').version
...
>>> print(f"pyceres 版本: {pyceres_version}")
pyceres 版本: 2.4

**Comment by songsenIng** [comment]
> Sorry, I try and I get the same error:
> 
> Python 3.9.20 | packaged by conda-forge | (main, Sep 30 2024, 17:49:10) 
> [GCC 13.3.0] on linux
> Type "help", "copyright", "credits" or "license" for more information.
> >>> import pycolmap
> >>> import pyceres
> >>> ba_options_tmp = pycolmap.BundleAdjustmentOptions()
> >>> ba_options_tmp.solver_options
> TypeError: Unregistered type : ceres::Solver::Options
> 
> The above exception was the direct cause of the following exception:
> 
> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
> TypeError: Unable to convert function return value to a Python type! The signature was
>         (self: pycolmap.BundleAdjustmentOptions) -> ceres::Solver::Options
> >>> pycolmap.__version__
> '3.10'
> >>> pyceres.__version__
> '2.2'
> >>>
I encountered the same problem as you. How did you explicitly import pyceres?Thank you！

```
(vggsfm) PS F:\vggsfm-main> python
Python 3.10.17 | packaged by conda-forge | (main, Apr 10 2025, 22:06:35) [MSC v.1943 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import pycolmap
>>> import pyceres
>>> ba_options_tmp = pycolmap.BundleAdjustmentOptions()
>>> ba_options_tmp.solver_options
TypeError: Unregistered type : ceres::Solver::Options

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: Unable to convert function return value to a Python type! The signature was
        (self: pycolmap.BundleAdjustmentOptions) -> ceres::Solver::Options
>>> exit()
(vggsfm) PS F:\vggsfm-main> 
```

**Comment by songsenIng** [comment]
> explicitly

Hi, I have the same problem, how can I  explicitly import it? Below is my message log. My environment is pycolmap==3.10.0 and pyceres==2.3

```
Traceback (most recent call last):
  File "F:\vggsfm-main\demo.py", line 71, in demo_fn
    predictions = vggsfm_runner.run(
  File "F:\vggsfm-main\vggsfm\runners\runner.py", line 229, in run
    predictions = self.sparse_reconstruct(
    ) = self.triangulator(
  File "E:\Anaconda_2023\envs\vggsfm\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "F:\vggsfm-main\vggsfm\models\triangulator.py", line 149, in forward
    ) = init_BA(
  File "F:\vggsfm-main\vggsfm\utils\triangulation.py", line 210, in init_BA
    ba_options = prepare_ba_options()
  File "F:\vggsfm-main\vggsfm\utils\triangulation_helpers.py", line 628, in prepare_ba_options
    ba_options_tmp.solver_options.function_tolerance *= 10
TypeError: Unable to convert function return value to a Python type! The signature was
        (self: pycolmap.BundleAdjustmentOptions) -> ceres::Solver::Options

Set the environment variable HYDRA_FULL_ERROR=1 for a complete stack trace.
(vggsfm) PS F:\vggsfm-main> python
Python 3.10.17 | packaged by conda-forge | (main, Apr 10 2025, 22:06:35) [MSC v.1943 64 bit (AMD64)] on win32
>>> import pycolmap
>>> import pyceres
>>> ba_options_tmp = pycolmap.BundleAdjustmentOptions()
>>> ba_options_tmp.solver_options
TypeError: Unregistered type : ceres::Solver::Options

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: Unable to convert function return value to a Python type! The signature was
        (self: pycolmap.BundleAdjustmentOptions) -> ceres::Solver::Options
>>> exit()
(vggsfm) PS F:\vggsfm-main> 
```

**Comment by MihailMihaylov97** [comment]
Are there any updates? I have the same issue with pycolmap == 3.10 and pyceres == 2.3.0

**Comment by ahojnnes** [comment]
This should be resolved with pycolmap==3.12.5. If not, please report back. Thank you.
